core/src/tile/tileWorker.cpp

```diff
@@ -78,26 +108,17 @@ void TileWorker::run() {
             m_queue.erase(it);
         }
 
-        if (task->isCanceled()) { continue; }
-
-        // Save shared reference to Scene while building tile
-        // FIXME: Scene could be released on Worker-Thread and
-        // therefore call unsafe glDelete* functions...
-        auto scene = m_scene;
-        if (!scene) { continue; }
+        if (task->isCanceled()) {
+            continue;
+        }
 
-        auto tileData = task->source().parse(*task, *scene->mapProjection());
+        auto tileData = task->source().parse(*task, *builder->scene().mapProjection());
 
         // const clock_t begin = clock();
 
-        context.initFunctions(*scene);
-
         if (tileData) {
-            auto tile = std::make_shared<Tile>(task->tileId(),
-                                               *scene->mapProjection(),
-                                               &task->source());
 
-            tile->build(context, *scene, *tileData, task->source());
+            auto&& tile = builder->build(task->tileId(), *tileData, task->source());
```

Comment:
It is true that in either case the `Tile` being pointed to will be neither constructed nor destructed, but creating a `shared_ptr` can have noticeable cost in tight loops, since they are made to be thread-safe. 

`auto tile = builder->...` would create a local `shared_ptr`. `std::move` transfers ownership of contents, but there would still be a `shared_ptr` on the stack that needs to be destructed (consider that it would still be legal to re-assign `tile` after the move).